### PR TITLE
Improvements to Jinja handling, formatting, output, and more

### PR DESCRIPTION
**Pull request by jpagh**
### Shared Jinja2 module (`_jinja.py`)
Extracted all Jinja2 logic into a new `src/dayamlchecker/_jinja.py` module so both
tools share the same detection, header validation, and pre-processing code:
- `_contains_jinja_syntax()` — regex-based detection of `{{ }}`, `{% %}`, `{# #}`
- `_has_jinja_header()` — checks for `# use jinja` on the first line
- `JinjaWithoutHeaderError` — raised when Jinja syntax is present without the header
- `_SilentUndefined` / `preprocess_jinja()` — renders Jinja templates with undefined
  variables silently resolved to empty values

### YAML checker (`yaml_structure`)
- `# use jinja` files are now **pre-processed through Jinja2** before structure
  checking, so the checker sees valid YAML instead of failing on template syntax
- Files without the `# use jinja` header that contain Jinja syntax produce an error
- CLI output: one character per file (`.` = ok, `j` = ok Jinja file); errors printed
  inline; trailing newline added so the shell prompt is never on the same line
- `--verbose` flag: for Jinja files, prints the rendered YAML that was actually checked

### Code formatter (`code_formatter`)
- `# use jinja` files are **no longer skipped**; instead, each `code:`/`validation
  code:` block is inspected individually — blocks containing Jinja syntax are left
  untouched while clean blocks are formatted with Black as normal
- CLI output: one character per file (`.` = unchanged, `R` = reformatted, `E` = error)
- Errors accumulated and printed to stderr after the progress line
- `--verbose` flag: prints a summary line (`N reformatted, N unchanged, N errors`)
- `--check` mode unchanged (still prints `Would reformat: <file>` per file)

### Bug fix: `DAFields` single-field dict shorthand
`fields:` written as a bare dict (single-field shorthand without a list wrapper) was
incorrectly rejected. It is now accepted if the dict contains at least one recognised
field key (`field`, `datatype`, `choices`, modifier keys, etc.).

### Tests
- 209 tests passing (up from ~100 at branch start)
- 100% coverage on `_jinja.py`
- New test classes for `YAMLStr`, `MakoText`, `PythonText`, `ObjectsAttrType`,
  `DAPythonVar`, `DAFields` (including shorthand fix), `ShowIf`, `YAMLError`,
  `find_errors`, `process_file`, CLI integration for both tools
- New formatter tests for `_reindent`, `_find_block_body_span`, `format_yaml_file`,
  `_collect_yaml_files`, Jinja mixed-block formatting, edge cases

### CODE REVIEW

**Review by BryceStevenWilley — CHANGES_REQUESTED** [review]
I'd appreciate it if you could split up the big features in this PR into their own PRs, i.e. a jinja PR, an output / extra CLI quiet args, that DAFields bug fix, PR, etc.

Otherwise it'll take me a while to review this

File: tests/test_code_formatter_cli.py [hunk 1]
```diff
@@ -46,3 +48,156 @@ def test_formatter_collect_yaml_files_can_disable_default_ignores():
...
 
         assert collected == sorted([visible, git_file, github_file, sources_file])
+
+
+def _run_formatter(*args: str) -> subprocess.CompletedProcess:
+    return subprocess.run(
+        [sys.executable, "-m", "dayamlchecker.code_formatter", *args],
+        capture_output=True,
+        text=True,
+    )
```
[the remark this answers is not in the document]
Comment (jpagh) [reply tests/test_code_formatter_cli.py]: Maybe I'm not understanding, but wouldn't that essentially NOT test the CLI/argparse?
